### PR TITLE
Opentel integration

### PR DESCRIPTION
**Pull request by elijahbenizzy**
Two modes:
1. Burr logs to OpenTel + Burr
2. OpenTel + Burr logs to Burr

TODO:
- [x] Clean up API
    - [x] Minimize classes/overlap
    - [x] Make it easy to use from a single tracker
    - [ ] Add registration for each integration point
- [x] Check thread safety with tracker reference from SpanProcessor
- [ ] Add documentation
    - [x] Class level
    - [ ] Concepts/references
- [x] Clean up code
- [x] Clean up/break into commits

Other thoughts:
- [ ] We need to be able to highlight specific spans for this to look good
- [ ] We need attributes somewhere else
- [ ] We want attribute aggregation (E.G. count) -- maybe some simple analytics view?

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

File: burr/integrations/opentelemetry.py [hunk 1]
```diff
@@ -66,21 +67,21 @@ def get_cached_span(span_id: int) -> Optional[FullSpanContext]:
 tracker_context = ContextVar[Optional[SyncTrackingClient]]("tracker_context", default=None)
 
 
-def _is_heterogeneous_array(value: list):
+def _is_heterogeneous_sequence(value: Sequence):
     if len(value) == 0:
         return True
     first_type = type(value[0])
-    return any([not isinstance(val, first_type) for val in value])
+    return all([isinstance(val, first_type) for val in value])
```
Comment (skrawcz) [inline burr/integrations/opentelemetry.py:74]: doesn't this need the negation? e.g. it's homogeneous with `all()` hetro if not.

File: tests/integrations/test_burr_opentelemetry.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "value, expected",
+    [
+        ("hello", "hello"),
+        (1, 1),
+        ((1, 1), [1, 1]),
+        ((1.0, 1.0), [1.0, 1.0]),
+        ((True, True), [True, True]),
+        (("hello", "hello"), ["hello", "hello"]),
+        (TestModel(foo=1, bar=True), json.dumps(serde.serialize(TestModel(foo=1, bar=True)))),
+    ],
```
Comment (skrawcz) [inline tests/integrations/test_burr_opentelemetry.py:25]: you don't actually have a hetero sequence here?

File: burr/integrations/opentelemetry.py (hunk 1 above)
Comment (elijahbenizzy) [reply burr/integrations/opentelemetry.py:74]: yeah the function is named poorly

File: tests/integrations/test_burr_opentelemetry.py (hunk 2 above)
Comment (elijahbenizzy) [reply tests/integrations/test_burr_opentelemetry.py:25]: again with the above

**Review by skrawcz — APPROVED** [review]

**Pull request merged**